tracer/src/Datadog.Trace/DuckTyping/DuckType.cs

```diff
@@ -988,15 +1129,15 @@ public bool CanCreate()
             }
 
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            internal object CreateInstance(object instance)
+            internal object? CreateInstance(object instance)
             {
-                return _activator.DynamicInvoke(instance);
+                return _activator?.DynamicInvoke(instance);
             }
 
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            private T ThrowOnError<T>(object instance)
+            private T? ThrowOnError<T>(object? instance)
             {
-                _exceptionInfo.Throw();
+                _exceptionInfo?.Throw();
                 return default;
```

Comment:
sure... but if this method is being called when `_exceptionInfo` is `null`, then that's an error in the duck typing infra right? This method, `ThrowOnError` _shouldn't_ ever return should it? Because you shouldn't be calling it if there isn't an error. 🤔 

Arguably we shouldn't have this method at all - we can probably refactor it so we either create or throw correctly, but I haven't checked, and that's definitely outside the scope of this PR.